### ISSUE TITLE
PEP 695 style type alias description

### ISSUE THREAD

**Issue by danpascu** [microsoft/pyright#8147]
I'm using pyright as a language server.

If I define a type alias like:

```py
type NSMap = dict[str | None, str]
```

then somewhere else in the code where the type is used for some annotation, if I hover over it, the description I get from pyright right now is:

```
(type alias) NSMap: type[TypeAliasType]
```

which is always the same, no matter what the alias maps to. The only thing different from one alias to another is just the alias name.

It would be nice if it could provide a more useful output, one that would include both the alias name and what is aliased to, like for example:

```
(type alias) NSMap: dict[str | None, str]
```

or maybe it could simply mirror the type definition:

```
type NSMap = dict[str | None, str]
```

That way I don't have to go to the alias definition to be able to see what it maps to. This is especially useful when dealing with some typing error and wanting to see how that type alias is defined.

**Comment by erictraut** [comment]
Thanks for the suggestion. I agree that when a type alias is used in a type annotation, hovering over that symbol should display the type associated with that type alias. When a type alias is used in a value expression, hovering over the symbol should display `TypeAliasType` (because that's the type of its value). 

```python
def func(x: NSMap): # Hovering over NSMap here should display "dict[str | None, str]`
    a = NSMap # Hovering over NSMap here should display "TypeAliasType"
```

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.368.

**Comment by danpascu** [comment]
Thank you.
